### PR TITLE
feat(frontend): support show internal tables

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

**This section will be used as the commit message. Please do not leave this empty!**

- Support show internal tables.

## Example:
```
dev=> show internal tables;
                   Name
------------------------------------------
 __internal_v_20_hashjoinright_1019
 __internal_v_20_hashjoindegreeleft_1018
 __internal_v_18_grouptopnnode_1011
 __internal_v_20_hashjoindegreeright_1020
 __internal_v_17_topnnode_1010
 __internal_v_19_hashjoindegreeleft_1014
 __internal_v_20_hashjoinleft_1017
 __internal_v_18_hashaggresult_1012
 __internal_v_19_hashjoinright_1015
 __internal_v_19_hashjoindegreeright_1016
 __internal_v_19_hashjoinleft_1013
(11 rows)

dev=> select * from __internal_v_19_hashjoinleft_1013 limit 5;
 orders.o_orderkey | orders.o_orderdate | orders.o_shippriority | customer.c_custkey | orders.o_custkey
-------------------+--------------------+-----------------------+--------------------+------------------
                69 | 1994-06-04         |                     0 |                 85 |               85
               256 | 1993-10-19         |                     0 |                125 |              125
              1154 | 1992-02-15         |                     0 |                 37 |               37
              1792 | 1993-11-09         |                     0 |                 49 |               49
              1894 | 1992-03-30         |                     0 |                 76 |               76
(5 rows)
```

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* SQL commands, functions, and operators

### Release note

- Support `show internal tables` statement. Developers will be able to explore the state implementation of RisingWave in depth with this statement.

## Refer to a related PR or issue link (optional)
https://github.com/risingwavelabs/risingwave/issues/7304

### CODE REVIEW

**Review by fuyufjh — APPROVED** [review]
LGTM

**Comment by jon-chuang** [comment]
Just curious, querying from this will query the `StateTable` via a `MaterializeExecutor` is this correct?

File: src/sqlparser/src/ast/mod.rs [hunk 1]
```diff
@@ -742,6 +743,9 @@ impl fmt::Display for ShowObject {
             ShowObject::Table { schema } => {
                 write!(f, "TABLES{}", fmt_schema(schema))
             }
+            ShowObject::InternalTable { schema } => {
```
Comment (jon-chuang) [inline src/sqlparser/src/ast/mod.rs:746]: Did we add to the AST tests?

**Comment by fuyufjh** [comment]
> Just curious, querying from this will query the `StateTable` corresponding to the internal table via a `MaterializeExecutor` is this correct?

No, it will query from storage layer directly. To be more precise, read from the shared buffer of the compute node(s)

File: src/sqlparser/src/ast/mod.rs (hunk 1 above)
Comment (jon-chuang) [reply src/sqlparser/src/ast/mod.rs:746]: https://github.com/risingwavelabs/risingwave/blob/main/src/sqlparser/tests/testdata/show.yaml

Comment (jon-chuang) [reply src/sqlparser/src/ast/mod.rs:746]: I think we can skip e2e test since it may be brittle, internal table may have a non-deterministic name...

**Pull request merged**